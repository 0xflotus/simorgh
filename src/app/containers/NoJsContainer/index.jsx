import React from 'react';
import Helmet from 'react-helmet';
import onClient from '#lib/utilities/onClient';

const NoJsContainer = () => {
  const htmlAttributes = {
    class: onClient() ? 'js' : 'no-js',
  };
  return <Helmet htmlAttributes={htmlAttributes} />;
};

export default NoJsContainer;
