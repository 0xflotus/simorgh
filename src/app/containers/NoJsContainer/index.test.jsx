/* eslint-disable jsx-a11y/html-has-lang */
import React from 'react';
import { mount, render } from 'enzyme';
import Helmet from 'react-helmet';
import NoJsContainer from '.';

describe('NoJsContainer', () => {
  it('should return a mounted component with the html class attribute as js', () => {
    const wrapper = mount(<NoJsContainer />);
    const { htmlAttributes } = Helmet.peek();
    expect(htmlAttributes.class).toBe('js');
    expect(wrapper).toMatchSnapshot();
  });

  it('should return the markup with the html class attribute as no-js', () => {
    render(<NoJsContainer />);
    const { htmlAttributes } = Helmet.peek();
    expect(htmlAttributes.class).toBe('no-js');
  });
});
