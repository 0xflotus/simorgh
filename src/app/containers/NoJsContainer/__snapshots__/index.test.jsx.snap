// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoJsContainer should return a mounted component with the html class attribute as js 1`] = `
<NoJsContainer>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    htmlAttributes={
      Object {
        "class": "js",
      }
    }
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      htmlAttributes={
        Object {
          "class": "js",
        }
      }
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        htmlAttributes={
          Object {
            "class": "js",
          }
        }
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</NoJsContainer>
`;
